pkg/storage/utils/eosfs/eosfs.go

```diff
@@ -913,6 +925,21 @@ func (fs *eosfs) createUserDir(ctx context.Context, u *userpb.User, path string,
 			return errors.Wrap(err, "eos: error setting attribute")
 		}
 	}
+
+	// set quota for user
+	quotaInfo := &eosclient.SetQuotaInfo{
+		Username:  u.Username,
+		MaxBytes:  fs.conf.DefaultQuotaBytes,
+		MaxFiles:  fs.conf.DefaultQuotaFiles,
+		QuotaNode: fs.conf.QuotaNode,
+	}
+
+	err = fs.c.SetQuota(ctx, uid, gid, quotaInfo)
```

Comment:
This will be called every time a reference is created, since that uses `createUserDir` as well. We can move this to `createNominalHome`.

Reply:
That also looks wrong, we should be creating the home on every reference.

Reply:
No we don't create a home directory. This is just a util method to create a directory whose owner is the user. It's used by multiple methods.

Reply:
@ishank011 take a look at last commit. Calling createUserDir will set xattrs on each reference that are already inherited from the shadow home. Just creating the directory and setting the ref is enough

Reply:
But would the user still be able to delete references? Given that they don't own those

Reply:
Looks good